maven-extension/src/main/java/io/opentelemetry/maven/handler/MojoGoalExecutionHandlerConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.maven.handler;
+
+import io.opentelemetry.maven.MavenGoal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.stream.Stream;
+
+public class MojoGoalExecutionHandlerConfiguration {
+
+  public static Map<MavenGoal, MojoGoalExecutionHandler> loadMojoGoalExecutionHandler(
+      ClassLoader classLoader) {
+
+    // built-in handlers
+    List<MojoGoalExecutionHandler> buildInHandlers =
+        Arrays.asList(
+            new GoogleJibBuildHandler(),
+            new MavenDeployHandler(),
+            new SnykMonitorHandler(),
+            new SnykTestHandler(),
+            new SpringBootBuildImageHandler());
+
+    List<MojoGoalExecutionHandler> spiHandlers = new ArrayList();
+    // Must use the classloader of the class rather the default ThreadContextClassloader to prevent
+    // java.util.ServiceConfigurationError:
+    //    io.opentelemetry.maven.handler.MojoGoalExecutionHandler:
+    //    io.opentelemetry.maven.handler.SpringBootBuildImageHandler not a subtype
+    ServiceLoader.load(MojoGoalExecutionHandler.class, classLoader)
+        .forEach(handler -> spiHandlers.add(handler));
+
+    Map<MavenGoal, MojoGoalExecutionHandler> mojoGoalExecutionHandlers = new HashMap<>();
+
+    Stream.concat(buildInHandlers.stream(), spiHandlers.stream())
+        .forEach(
+            handler ->
+                handler
+                    .getSupportedGoals()
+                    .forEach(
+                        goal -> {
+                          MojoGoalExecutionHandler previousHandler =
+                              mojoGoalExecutionHandlers.put(goal, handler);
```

Comment:
Think you can use `toMap` since it will throw on duplicate keys

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-

Reply:
Thanks @anuraaga . Could you please give me more guidance on this one? 
I don't catch how I can use `stream.collect(Collectors.toMap())` as I have a double loop 👼

Reply:
Oops - yeah didn't think enough about it, stream doesn't seem good for this case